Update README to fix bugs, add links, and update structure (#3371)
returntocorp/sgrep-rules: README.md

# semgrep-rules

[![powered by semgrep](https://img.shields.io/badge/powered%20by%20semgrep-2ACFA6)](https://semgrep.dev/)
<a href="https://go.semgrep.dev/slack">
<img src="https://img.shields.io/badge/community%20slack-3.5k%20members-green?style=flat-square" alt="Join Semgrep community Slack" />
</a>

Welcome! This repository is the standard library for open source [Semgrep](https://semgrep.dev/) rules.

> In addition to rules in this repository, the [Semgrep Registry](https://semgrep.dev/explore) also includes proprietary rules that enable interfile and interprocedural analsis, called [Pro rules](https://semgrep.dev/products/semgrep-code/pro-rules).

Reviewer: ```suggestion
In addition to the rules in this repository, the [Semgrep Registry](https://semgrep.dev/explore) offers proprietary [Pro rules](https://semgrep.dev/products/semgrep-code/pro-rules) that enable interfile and interprocedural analysis.
```

- Find rules: search for open source and Pro rules through the [Semgrep registry search](https://semgrep.dev/r).

> - Use rules: Scan your code with these rules through the [Semgrep AppSec Platform](https://semgrep.dev/login)

Reviewer: ```suggestion
- Use rules: Scan your code with these rules through [Semgrep AppSec Platform](https://semgrep.dev/login)
```

> - Contribute to rules: see the [Contributing to Semgrep rules](https://semgrep.dev/docs/contributing/contributing-to-semgrep-rules-repository/) documentation.

Reviewer: ```suggestion
- Contribute to rules: see [Contributing to Semgrep rules](https://semgrep.dev/docs/contributing/contributing-to-semgrep-rules-repository/) for more information.
```

## Using the Semgrep rules repository

> Run existing and custom Semgrep rules locally with the Semgrep command line interface (Semgrep CLI) or continuously with Semgrep in CI while using Semgrep AppSec Platform. To start using Semgrep rules, see [Semgrep tutorial](https://semgrep.dev/learn), [Getting started with Semgrep CLI](https://semgrep.dev/docs/getting-started/), and [Getting started with Semgrep App](https://semgrep.dev/docs/semgrep-app/getting-started-with-semgrep-app/). To write your own Semgrep rules, see the [Semgrep tutorial](https://semgrep.dev/learn) and [documentation on writing rules](https://semgrep.dev/docs/writing-rules/overview/).

Reviewer: ```suggestion
To start writing and using Semgrep rules, see [Learn Semgrep syntax](https://semgrep.dev/learn) and [Writing rules](https://semgrep.dev/docs/writing-rules/overview/). Then, run existing and custom Semgrep rules locally with the [Semgrep command line interface (Semgrep CLI)](https://semgrep.dev/docs/getting-started/) or [continuously with Semgrep in CI while using Semgrep AppSec Platform](https://semgrep.dev/docs/semgrep-app/getting-started-with-semgrep-app/).
```

## Writing Semgrep rules

> See Semgrep documentation on [writing rules](https://semgrep.dev/docs/writing-rules/overview/), including:

Reviewer: ```suggestion
See [Writing rules](https://semgrep.dev/docs/writing-rules/overview/) for information including:
```

> - Pattern syntax, describing what Semgrep patterns can do in detail, and provides example use cases of the ellipsis operator, metavariables.

Reviewer: ```suggestion
- Pattern syntax, describing what Semgrep patterns can do in detail, and example use cases of the ellipsis operator, metavariables.
```
